assets/js/modules/analytics-4/datastore/properties.js

```diff
@@ -717,27 +707,33 @@ const baseActions = {
 
 const baseControls = {};
 
-function baseReducer( state, { type, payload } ) {
-	switch ( type ) {
-		case MATCHING_ACCOUNT_PROPERTY:
-			return { ...state, ...payload };
-		case SET_HAS_MISMATCHED_TAG:
-			return {
-				...state,
-				moduleData: {
-					...state.moduleData,
-					hasMismatchedTag: payload.hasMismatchedTag,
-				},
-			};
-		case SET_IS_WEBDATASTREAM_AVAILABLE:
-			return {
-				...state,
-				isWebDataStreamAvailable: payload.isWebDataStreamAvailable,
+const baseReducer = createReducer( ( state, action ) => {
+	switch ( action.type ) {
+		case MATCHING_ACCOUNT_PROPERTY: {
+			const { isMatchingAccountProperty } = action.payload;
+			state.isMatchingAccountProperty = isMatchingAccountProperty;
+			break;
+		}
+
+		case SET_HAS_MISMATCHED_TAG: {
+			const { hasMismatchedTag } = action.payload;
+			state.moduleData = {
```

Comment:
This can use the same pattern as elsewhere:

```js
			state.moduleData = state.moduleData || {};
			state.moduleData.hasMismatchedTag = payload.hasMismatchedTag;
```